BUG: Non-central Fisher distribution (fix nan-values when nc=0)

scidam opened the pull request:
#### Reference issue
Closes gh-11660

#### What does this implement/fix?
Non-central Fisher distribution (`ncf`) doesn't work for nc=0 and return nan-values.
Adding custom argument checking method to the class `gen_ncf` resolves the problem.

#### Additional information
 No additional information.

chrisb83 commented:
Thanks for submitting this PR. The fix makes sense but the documentation needs to be adjusted:

- docstring of the function (it incorrectly states n1 > 1 and lambda > 0)
- tutorial (https://docs.scipy.org/doc/scipy/reference/tutorial/stats/continuous_ncf.html) says lambda > 0

optional: add see also and link F distribution

scidam commented:
@chrisb83, I've updated the documentation as you suggested. Also, I found that in the tutorial doc (about ncf) both  `\nu_1`/ `\nu_2` and `v_1`/ `v_2`  are used in the same context [18b6346](https://github.com/scipy/scipy/pull/11667/commits/18b6346cc874c228f822e6242056f6ddfa21de84). I replaced `v_{i}` with `\nu_{i}`.

mdhaber commented:
I re-triggered the Windows builds on Azure; not sure why they exceeded the time limit.
The Travis errors are unrelated.

scidam commented:
The same issue took place, e.g., PR-11684, and others. It is unbelivable, if this timeout issue is caused by the tiny code changes I made.

tylerjereddy commented:
The time limit thing is happening more often lately with Azure. That 32-bit Windows job tends to fail anyway with another test problem for `optimize` lately. Branch needs a rebase to remove the Travis issues.

scidam commented on scipy/stats/tests/test_distributions.py:
```diff
@@ -4145,6 +4145,25 @@ def test_crystalball_function_moments():
+    x = np.array([-0.5, 0.2, 1.0, 2.3])
+    df1 = 4
+    df2 = 11
+    nc = 0
```
Fixed

scidam commented on scipy/stats/_continuous_distns.py:
```diff
@@ -5748,19 +5748,27 @@ class ncf_gen(rv_continuous):
 
-    `ncf` takes ``df1``, ``df2`` and ``nc`` as shape parameters.
+    `ncf` takes ``df1``, ``df2`` and ``nc`` as shape parameters. If ``nc=0``,
+    it becomes equivalent to the Fisher distribution.
```
Fixed

scidam commented on doc/source/tutorial/stats/continuous_ncf.rst:
```diff
@@ -19,4 +19,7 @@ There are 3 shape parameters: the degrees of freedom :math:`\nu_{1}>0` and :math
 
 where :math:`L_{\nu_{2}/2}^{\nu_{1}/2-1}(x)` is an associated Laguerre polynomial.
 
+If :math:`\lambda=0`, it becomes equivalent to the Fisher distribution
```
Fixed

scidam commented on scipy/stats/_continuous_distns.py:
```diff
@@ -5748,19 +5748,27 @@ class ncf_gen(rv_continuous):
                               {B(v_1/2, v_2/2)  \gamma(\frac{v_1+v_2}{2})}
 
-    for :math:`n_1 > 1`, :math:`n_2, \lambda > 0`.  Here :math:`n_1` is the
+    for :math:`n_1, n_2 > 0`, :math:`\lambda\geq0`.  Here :math:`n_1` is the
```
Take a look at "compiled" version of the [docs](https://docs.scipy.org/doc/scipy/reference/tutorial/stats/continuous_ncf.html) about non-central Fisher distribution. In the first line of the text: there are both used \nu_1, \nu_2 and v_1 and v_2 in the same context. This is typo, I think. I just suggested to use \nu_i instead of wrong v_1 and v_2 within this document only.

rlucas7 replied:
ah I see now, they are 2 different docs, 1 is the tutorial, where the type was, that explains, thanks.

scidam commented on doc/source/tutorial/stats/continuous_ncf.rst:
```diff
@@ -5,10 +5,10 @@ Noncentral F Distribution
+and parameter :math:`\lambda`, and :math:`X_{2}` is chi-squared with :math:`\nu_{2}` degrees of freedom.
 
-There are 3 shape parameters: the degrees of freedom :math:`\nu_{1}>0` and :math:`\nu_{2}>0`; and :math:`\lambda>0`.
+There are 3 shape parameters: the degrees of freedom :math:`\nu_{1}>0` and :math:`\nu_{2}>0`; and :math:`\lambda\geq0`.
```
Fixed

WarrenWeckesser commented:
Squashed and merged.  @scidam, thanks for the fix!
